### PR TITLE
Support Hive get_json_object to Presto json_extract

### PR DESCRIPTION
**Pull request by ebyhr**

### CODE REVIEW

**Review by wmoustafa — APPROVED** [review]
Thanks @ebyhr for this patch. LGTM.

File: coral-presto/src/test/java/com/linkedin/coral/presto/rel2presto/HiveToPrestoConverterTest.java [hunk 1]
```diff
@@ -80,6 +80,8 @@ public void testViews(String database, String view, String expectedSql) {
             + "SELECT \"a\", CAST(row(b.b1, transform_values(b.m1, (k, v) -> cast(row(v.b1, transform(v.a1, x -> cast(row(x.b1) as row(b1 varchar)))) as row(b1 varchar, a1 array(row(b1 varchar)))))) as row(b1 varchar, m1 map(varchar, row(b1 varchar, a1 array(row(b1 varchar)))))) AS \"b\"\nFROM \"test\".\"tablen\"\nUNION ALL\n"
             + "SELECT \"a\", \"b\"\n" + "FROM \"test\".\"tableo\")" },
 
-        { "test", "current_date_and_timestamp_view", "SELECT CURRENT_TIMESTAMP, TRIM(CAST(CURRENT_TIMESTAMP AS VARCHAR(65535))) AS \"ct\", CURRENT_DATE, CURRENT_DATE AS \"cd\", \"a\"\nFROM \"test\".\"tablea\"" } };
+        { "test", "current_date_and_timestamp_view", "SELECT CURRENT_TIMESTAMP, TRIM(CAST(CURRENT_TIMESTAMP AS VARCHAR(65535))) AS \"ct\", CURRENT_DATE, CURRENT_DATE AS \"cd\", \"a\"\nFROM \"test\".\"tablea\"" },
+
+        { "test", "get_json_object_view", "SELECT \"json_extract\"(\"b\".\"b1\", '$.name')\nFROM \"test\".\"tablea\"" } };
```
Comment (findepi) [inline coral-presto/src/test/java/com/linkedin/coral/presto/rel2presto/HiveToPrestoConverterTest.java:85]: Won't such a view  cause

```
Query failed View 'hive.v' is stale or in invalid state: column [a] of type json projected from query view at position 1 cannot be coerced to column [a] of type varchar stored in view definition
```
?

@ebyhr can you please also add a test on Trino side, in `TestHiveViews`?

Comment (ebyhr) [reply coral-presto/src/test/java/com/linkedin/coral/presto/rel2presto/HiveToPrestoConverterTest.java:85]: > Won't such a view cause?

Confirmed this exactly causes the error in Trino side.

Comment (wmoustafa) [reply coral-presto/src/test/java/com/linkedin/coral/presto/rel2presto/HiveToPrestoConverterTest.java:85]: @findepi, could you clarify when such errors happen on the Trine side? I see a follow up on this in https://github.com/trinodb/trino/issues/5688.

Comment (ebyhr) [reply coral-presto/src/test/java/com/linkedin/coral/presto/rel2presto/HiveToPrestoConverterTest.java:85]: @wmoustafa This is the steps to reproduce. Even simple view causes the issue.
```
hive> create table test_base (c1 string);
hive> create view test_view as select get_json_object(c1, "$.name") from test_base;
hive> create view test_view2 as select get_json_object(c1, "$.name") as c2 from test_base;

trino> select * from hive.default.test_view;
View 'hive.default.test_view' is stale or in invalid state: a column of type json projected from query view at position 0 has no name

trino> select * from hive.default.test_view2;
View 'hive.default.test_view2' is stale or in invalid state: column [c2] of type json projected from query view at position 0 cannot be coerced to column [c2] of type varchar stored in view definition
```

**Comment by wmoustafa** [comment]
Thanks @ebyhr! I was wondering how @findepi predicated this would happen on the Trino side.

**Comment by findepi** [comment]
> was wondering how @findepi predicated this would happen on the Trino side

Crystal ball. :) 

(i hit this or a related problem when working on json_tuple)

**Pull request merged**